src/server/general/creation.md

```diff
@@ -0,0 +1,99 @@
+# Creating the Display and event loops
+
+wayland-server provides a way to instantiate a `Display` for use by the compositor. Simply calling the `new` function creates a display. However this display is not exposed to clients. To allow clients to connect to the `Display`, a socket needs to be created. A socket may be created through the [`Display::add_socket_auto`] which creates a socket using the first available name.
+
+For compositors which wish to have more control, you may also specify the name of the socket to use with the [`Display::add_socket`], [`Display::add_socket_fd`] or [`Display::add_socket_from`] functions.
+
+```rust,norun
+use std::env;
+use wayland_server::Display;
+
+let display = Display::new();
+
+let socket_name = display
+    .add_socket_auto()
+    .expect("Failed to add wayland socket")
+    .into_string()
+    .unwrap();
+
+println!("Listening on wayland socket {}", socket_name);
+env::set_var("WAYLAND_DISPLAY", &socket_name);
+```
+
+At this point a client can attempt to connect to the compositor. However you may notice using [wayland-info](https://gitlab.freedesktop.org/wayland/wayland-utils) that when running wayland-info, the application hangs with no response. The cause of this behavior is that the compositor's `Display` does not respond to any requests from the client and wayland-info will wait until it receives a response.
+
+## The event loop
+
+An event loop is the primary way Wayland compositors are driven. Smithay uses [calloop](https://github.com/Smithay/calloop) to listen to the file descriptor created by the the `Display`. 
+
+A `Display` is driven through an `EventLoop` and is inserted as an event source through the event loop's handle. Callbacks are applied when the event loop dispatches events when a source has generated some new events.
+
+Since we will need to provide the `Display` inside the callback to dispatch events, we will need to wrap the `Display` inside an `Rc<RefCell<_>>`. To further simplify future expansion, also create a structure to store state about the compositor:
+
+```rust,no_run
+use std::{cell::RefCell, env, rc::Rc, sync::{Arc, atomic::AtomicBool};
+use wayland_server::Display;
+
+struct State {
+    display: Rc<RefCell<Display>>,
+    /// Tracks whether the compositor should continue to run.
+    running: Arc<AtomicBool>,
+}
+```
+
+You will also need to create an event loop:
+
+```rust,no_run
+let mut event_loop = EventLoop::try_new().unwrap();
```

Comment:
I think it would be nice to somehow let the reader know that this is `calloop::EventLoop` possibly by adding an use statement. Same for `calloop::LoopHandle`. It may be obvious for as that those come from `calloop`, but if someone is completely new, they might be searching for those types in dosc.rs/smithay instead of docs.rs/calloop

Reply:
Might be worth noting this inside of a sort of conventions document for the book repo?